### PR TITLE
Fix #1528: Add STATUS_REASON to operations for Cancel of the operation

### LINKED ISSUE #1528: Add STATUS_REASON to operations for Cancel of the operation

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/OperationEntity.java [hunk 1]
```diff
@@ -36,6 +38,7 @@
  */
 @Entity
 @Table(name = "pa_operation")
+@Getter @Setter
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/database/model/entity/OperationEntity.java:41]: I will covert other entities into Lombok in a separate PR.

**Review by petrdvorak — DISMISSED** [review]
Looks OK

File: powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/OperationApproveRequest.java [hunk 2]
```diff
@@ -43,4 +45,11 @@ public class OperationApproveRequest {
     @JsonSetter(nulls = Nulls.SKIP)
     private final Map<String, Object> additionalData = new LinkedHashMap<>();
 
+    /**
+     * Optional details why the status has changed. The value is more about code than free-text detail.
+     */
+    @Schema(description = "Optional details why the status has changed. The value is more about code than free-text detail.")
+    @Size(max = 32)
+    private String statusReason;
```
Comment (banterCZ) [inline powerauth-client-model/src/main/java/com/wultra/security/powerauth/client/model/request/OperationApproveRequest.java:53]: - [ ] Let's keep `statusReason` only for the cancel request.

File: docs/Database-Structure.md [hunk 3]
```diff
@@ -542,6 +543,7 @@ CREATE TABLE pa_operation (
 | data                | text         | -           | Data of the operation that enter the final signature.                                           |
 | parameters          | text         | -           | JSON-encoded parameters that were used while creating the operation.                            |
 | status              | integer      | -           | Status of the operation.                                                                        |
+| status_reason       | varchar(32)  | -           | Optional details why the status changed. The value is more about code than free-text detail.    |
```
Comment (romanstrobl) [inline docs/Database-Structure.md:546]: Can we update the second sentence?

File: docs/PowerAuth-Server-1.8.0.md [hunk 4]
```diff
@@ -24,3 +24,10 @@ transport hints to a FIDO2 Authenticator registered in the table. These transpor
 credential list or exclude credential list during WebAuthn ceremonies, serving as a fallback if the client fails to
 provide transport hints when registering a new credential. The format of the column is a list of authenticator transport
 values supported by the WebAuthn protocol, serialized as a JSON array.
+
+
+### Add status_reason Column
+
+A new column `status_reason` has been added to the `pa_operation` table.
+It provides optional details why the status changed.
+The value is more about code than free-text detail.
```
Comment (romanstrobl) [inline docs/PowerAuth-Server-1.8.0.md:33]: Can we update this sentence?

**Review by romanstrobl — APPROVED** [review]
I found several minor wording issues.

**Review by petrdvorak — APPROVED** [review]

**Pull request merged**